### PR TITLE
Complete

### PR DESCRIPTION
**Pull request by jimsch**

### CODE REVIEW

File: include/cn-cbor/cn-cbor.h [hunk 1]
```diff
@@ -388,10 +440,26 @@ cn_cbor* cn_cbor_array_create(CBOR_CONTEXT_COMMA cn_cbor_errback *errp);
...
                           cn_cbor* cb_value,
                           cn_cbor_errback *errp);
 
+/**
+ * Dump the object to a file pointer
+ * If buffer is NULL, then return required size to generate output
+ *
+ * @param[in]   buffer	Location to place output
+ * @param[in]   bufferSize Size of return buffer
+ * @param[in]	fp		File pointer to print on
```
Comment (sbertin-telular) [inline include/cn-cbor/cn-cbor.h]: There is no parameter fp.

File: src/cn-cbor.c [hunk 2]
```diff
@@ -158,7 +165,7 @@ static cn_cbor *decode_item (struct parse_buf *pb CBOR_CONTEXT, cn_cbor* top_par
     break;
   case MT_BYTES: case MT_TEXT:
     cb->v.str = (char *) pos;
-    cb->length = val;
+    cb->length = (size_t) val;
```
Comment (sbertin-telular) [inline src/cn-cbor.c]: This type cast is only correct if _MSC_VER is defined.

File: src/cn-print.c [hunk 3]
```diff
@@ -0,0 +1,253 @@
+#ifndef CN_PRINT_C
+#define CN_PRINT_C
+#define CN_INCLUDE_DUMPER
+#ifdef CN_INCLUDE_DUMPER
+#define _CRT_SECURE_NO_WARNINGS 1
+
+#include <stdio.h>
+#include <stdio.h>
```
Comment (sbertin-telular) [inline src/cn-print.c]: stdio.h is included twice here and once below. Probably only need it once.

**Comment by jimsch** [comment]
I am trying to figure out what needs to be dealt with in trying to merge all of the new updates in the cabo master.  I am not sure when this is going to actually be pulled as it all needs to get dealt with.

**Comment by jimsch** [comment]
Try and do this a different way

**Pull request closed without merging**